src/NativePresets/CMakeLists.txt

```diff
@@ -0,0 +1,70 @@
+if(NOT ENABLE_PRESETS OR NOT ENABLE_NATIVE_PRESETS)
+    return()
+endif()
+
+# Using a macro to reduce repetition.
+macro(add_native_preset name)
+    if(NOT CMAKE_SYSTEM_NAME STREQUAL "Windows")
+        add_library(${name}_static STATIC ${name}.cpp)
+
+        set_target_properties(${name}_static PROPERTIES
+                OUTPUT_NAME ${name}
+                )
+
+        target_link_libraries(${name}_static
+                PRIVATE
+                projectM_static
+                GLM::GLM
+                )
+    
+        install(TARGETS ${name}_static
+                LIBRARY DESTINATION "${PROJECTM_DATADIR_PATH}/presets"
+                RUNTIME DESTINATION "${PROJECTM_DATADIR_PATH}/presets"
+                ARCHIVE DESTINATION "${PROJECTM_DATADIR_PATH}/presets"
+                COMPONENT Presets
+                )
+    endif()
+
+    add_library(${name}_shared SHARED ${name}.cpp)
+
+    target_compile_definitions(${name}_shared
+            PRIVATE
+            ${name}_EXPORTS
+            )
+
+    set_target_properties(${name}_shared PROPERTIES
+            OUTPUT_NAME ${name}
+            )
+
+    # On Windows, libprojectM is only built statically.
```

Comment:
[WINDOWS_EXPORT_ALL_SYMBOLS](https://cmake.org/cmake/help/latest/prop_tgt/WINDOWS_EXPORT_ALL_SYMBOLS.html) can help with this.

Reply:
Kind of a brute-force solution though. As of now, the library has no really well-defined export interface, the installed headers are not exporting all functionality the library could supply and having all symbols exported will both increase the symbol table size and make it harder to later clean up the API or change any code if people start using all the internal classes.

This is a bit out of scope of simply adding a new build system and will require some work on the codebase. I kept it the way the original build files (the VS solution in this case) handled the build.

Reply:
If you want to put in the work to use Microsoft's nonstandard macros, sure go ahead. Personally I like CMake because it requires little work to support operating systems I don't use myself.

Reply:
I'd use CMake's [export header generator](https://cmake.org/cmake/help/latest/module/GenerateExportHeader.html) module that creates a header with wrapper macros tailored to the compiler being used. For example this will use `declspec _dllexport` on MSVC and `visibility(visible)` and setting default symbol visibility to hidden on UNIX compilers like GCC and Clang. Really neat.